### DISCUSSION TITLE
Input loop not working when defining multiple media in egs_chamber

### DISCUSSION THREAD

**Discussion by nexustowers** [nrc-cnrc/EGSnrc#1237] [Questions] [answered]
I'm trying to use an input loop using a string variable in egs_chamber to define multiple materials using the same density correction file by using the following placed inside the media definition block -

```
:start input loop:
   loop count = 2
   loop variable = 2 myvar water1 water2
       :start $(myvar):
            density correction file = water_liquid
       :stop $(myvar):
:stop input loop:
```

When I run the application it seems the loop isn't working though - I get the error message "Material not defined: water1"

I've never had a problem with input loops before, but in this case I'm guessing the variable names aren't able to be passed to text surrounded with colons?
It's pegsless mode and I am including the ae, ap etc definitions

**Comment by rtownson** [comment, marked as answer]
I'm guessing this is because the media definition section of the input file is actually parsed by the mortran code (because you can have the same media input block in the mortran apps). This should at the very least be in the documentation as a known issue..

**Comment by nexustowers** [comment]
Thanks very much for the reply.  Are there any other definition sections of the input file which might not work due to being parsed by Mortran?

**Reply by rtownson** [reply]
Just the transport parameters.

**Comment by nexustowers** [comment]
ok, thank you
